tests/utils.py

```diff
@@ -61,3 +65,34 @@ def get_node_name(nodename, ssh_config=None):
                 'salt-call --local --out txt grains.get id | cut -c 8-'
             )
     return nodename
+
+
+# Source: https://www.peterbe.com/plog/best-practice-with-retries-with-requests
+
+def requests_retry_session(
+    retries=3,
+    backoff_factor=0.3,
+    status_forcelist=(500, 503),
+    method_whitelist=frozenset(['GET', 'POST']),
+    session=None
+):
+    """Request library retry helper
+
+    This helper performs 3 retries with an exponential sleep interval between
+    each request.
+
+    We only retry internal server errors(500) & service unavailable errors(503)
```

Comment:
This is the default behavior (so you could start the sentence with "By default this helper…"

Also would be nice to have doc for the arguments, something like that:

```python
    """Configure a `requests.Session` for retry on error.

    By default this helper performs 3 retries with an exponential sleep interval
    between each request and only retry internal server errors(500) & service
    unavailable errors(503).

    Arguments:
        retries:          […]
        backoff_factor:   […]
        status_forcelist: […]
        method_whitelist: […]
        session:          […]

    Returns:
        A `requests.Session` object configured for retry.
    """
```

Reply:
Done